### PR TITLE
fix icon selection

### PR DESCRIPTION
**Pull request by dirk-thomas**
Currently the package tries to load the icon in the query view based in the full slug name. For many languages which have slugs for several version numbers there is no icon with such a name. And it would be "bad" to duplicate the icons for the various variations just to match the slug name.

Therefore this patch looks at all the available icon files and try to match:
* the full slug name (e.g. `cmake~3.5`)
* then the variation without the minor version after the `.` (e.g. `cmake~3`)
* then the variation without the version after the `~` (e.g. `cmake`)

The most specific match will be used. Commonly there are only icons without a version number. But this allows to use different icons when necessary.

If no matching icon is found the full slug name is being used (as before). Instead the logic could also fallback to a default icon if that would be desired.

This fixes at least the icon problem mentioned in #29.

### CODE REVIEW

File: src/query_view.js [hunk 1]
```diff
@@ -56,6 +60,47 @@ class QueryView extends SelectListView {
...
+      if (Icons.indexOf(name) != -1) {
+        return name;
+      }
+    }
+    // fallback to the full slug (even if it doesn't exist)
+    // optionally this could show a default icon?
+    return slug;
+  }
+
+  lazyLoadIcons_() {
```
Comment (sharvil) [inline src/query_view.js]: I understand the desire to bring all icons in memory for quick lookups in the future. That said, I think this won't scale well and might be better without caching or with a fixed-size, demand-loaded cache instead.

Comment (dirk-thomas) [inline src/query_view.js]: This only "loads" the filenames of the available icons. This is only 65 string which on average are shorter then 10 characters. I think this is much better in performance than to do a file system access every time the view if being generated.

Comment (sharvil) [inline src/query_view.js]: Agreed. Maybe you can rename the function / variable to indicate that only file names are being cached and not the icons themselves. Skimming through the code, it looks like there's an entire icon cache and not just a file name cache.

Comment (dirk-thomas) [inline src/query_view.js]: Good point. I have renamed the variable to `IconNames` and the method to `lazyLoadIconNames_`.

**Comment by dirk-thomas** [comment]
@sharvil I would like to go ahead and do a patch release with the recent fixes if that ok with you?

**Comment by dirk-thomas** [comment]
I have released version 0.0.5. I hope that is fine.

**Comment by sharvil** [comment]
Sounds good, thanks!

**Pull request merged**